sale_rental/models/product.py

```diff
@@ -81,5 +81,5 @@ def _inverse_rented_product_tmpl_id(self):
         for template in self:
             if len(template.product_variant_ids) == 1:
                 template.product_variant_ids.rented_product_id = (
-                    template.rented_product_tmpl_id.product_variant_ids[0].id
+                    template.rented_product_tmpl_id.product_variant_ids[:1].id
```

Comment:
```suggestion
                    from fields import first
                    first(template.rented_product_tmpl_id.product_variant_ids).id
```

This is better

Reply:
Yes, but you are in Odoo context. That's common in Odoo

Reply:
[:1] is common in Odoo

Reply:
It's not about which one is more common, it's about the API dictating in which way you get the first record from a recordset; and since they provide an interface for that, it is reasonable to use it.

They may offer some sort of lazy loading in the feature that will be used by that API.

I do agree though that it's location is odd; I would expect something of `template.rented_product_tmpl_id.product_variant_ids.first()`